dbms/src/Storages/DeltaMerge/DMContext.cpp

```diff
@@ -67,6 +68,11 @@ DMContext::DMContext(
     , read_stable_only(settings.dt_read_stable_only)
     , enable_relevant_place(settings.dt_enable_relevant_place)
     , enable_skippable_place(settings.dt_enable_skippable_place)
+#if ENABLE_NEXT_GEN
+    , fap_use_segment_to_end_map_cache(settings.fap_use_segment_to_end_map_cache)
+#else
+    , fap_use_segment_to_end_map_cache(true)
+#endif
```

Comment:
@CalvinNeo PTAL, can we remove the `settings.fap_use_segment_to_end_map_cache` and skip the segment_to_end_map_cache by default?

Reply:
Yes, skip it is fine. We can switch to https://github.com/tidbcloud/tiflash-cse/pull/387